### PR TITLE
Use azure temp storage for kitchen tests

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?
Use temp storage on azure VMs to copy test files over since we're running out of space on `/tmp` on these VMs. The temp storage is mounted at `/mnt`.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]

File: .gitlab/functional_test/system_probe.yml [hunk 1]
```diff
@@ -76,7 +76,6 @@ kitchen_test_system_probe_windows_x64:
   variables:
     KITCHEN_ARCH: x86_64
     KITCHEN_OSVERS: "win2012r2"
-    CHEF_VERSION: 14.12.9
```
Comment (albertvaka) [inline .gitlab/functional_test/system_probe.yml:79]: Is this change wanted? What version is this going to use by default?

**Review by albertvaka — APPROVED** [review]
The changes look good to me but I don't understand why we are removing `CHEF_VERSION`. Can you explain it before merging?

File: .gitlab/functional_test/system_probe.yml (hunk 1 above)
Comment (KSerrania) [reply .gitlab/functional_test/system_probe.yml:79]: The version defined in `.kitchen_test_system_probe`: `15.16.4`

Comment (hmahmood) [reply .gitlab/functional_test/system_probe.yml:79]: It is required for the `azure?` check here: https://github.com/DataDog/datadog-agent/pull/11071/files#diff-f3d6afcce67313e76a58c652295befb80079ecff110015a64fa5a62e99ff6d5aR8

**Pull request merged**